include/message_data_fmt_nes.h

```diff
@@ -0,0 +1,340 @@
+#ifndef MESSAGE_DATA_FMT_NES_H
+#define MESSAGE_DATA_FMT_NES_H
+
+/*
+ * Macros to create both a constant and a string literal from a magic value
+ *   The constants are used in code files when parsing text for various purposes
+ *   The strings are used in the message_data_static files themselves, as you can only concat strings with other strings
+ */
+
+#ifndef GLUE
+#define GLUE(a, b) a##b
+#endif
+
+#define STRINGIFY(s) #s
+#define EXPAND_AND_STRINGIFY(s) STRINGIFY(s)
+
+#define HEX(N) GLUE(0x, N)
```

Comment:
Glue?

Reply:
It glues the two arguments together into 1 token. For example `HEX(10) = GLUE(0x, 10) = 0x10`.